creusot/src/backend/clone_map/elaborator.rs

```diff
@@ -330,26 +329,26 @@ impl DepElab for TyElab {
         match ty.kind() {
             TyKind::Alias(_, _) => vec![ctx.assoc_ty_decl(&mut names, def_id, subst)],
             _ => {
-                if let Some(why3_modl) = util::get_builtin(ctx.tcx, def_id) {
-                    let qname = QName::from_string(why3_modl.as_str());
-                    let Kind::Used(_, _) = names.insert(Dependency::Type(ty)) else {
-                        return vec![];
-                    };
-
-                    let use_decl = Use { as_: None, name: qname.module_qname(), export: false };
-
-                    vec![Decl::UseDecl(use_decl)]
-                } else {
-                    let name = names.insert(dep).qname();
-
-                    let modl = if util::item_type(ctx.tcx, def_id) == ItemType::Closure {
-                        ctx.module_path_with_suffix(def_id, "_Type")
-                    } else {
-                        ctx.module_path(def_id)
-                    };
-                    let name = if name.module.is_empty() { name } else { name.module_qname() };
-                    let use_decl = Use { as_: Some(name.as_ident()), name: modl, export: false };
-                    vec![Decl::UseDecl(use_decl)]
+                match names.insert(dep) {
+                    Kind::Unnamed | Kind::Named(_) => {
+                        // Locally available names.
+                        // We return an empty vector because we have to return something,
+                        // but it will be ignored anyway: it is inserted into a map at a
+                        // key which will never be looked up.
+                        vec![]
```

Comment:
It would be good to figure out which types are causing this, but not a blocker. Normally everything in the graph here should be connected and thus used eventually...